scrapbook-tutorial/src/App.js

```diff
@@ -1,26 +1,110 @@
-import React, { Component } from "react";
+import React, { useState, useEffect, useCallback } from "react";
 import Comment from "./Comment";
 import Form from "./Form";
+import Dialog from "./Dialog";
+import axios from "axios";
+import ApiList from "./ApiList";
+import { DragDropContext, Droppable } from "react-beautiful-dnd";
 
-class App extends Component {
-  state = {
-    characters: [],
+const myDogServerBaseURL = "https://dog.ceo/api";
+
+const App = () => {
+  var previousActiveElement;
+  var dialog;
+  var dialogMask;
+  var dialogWindow;
+
+  var lastFocusableEl;
+  var firstFocusableEl;
+
+  const [comments, setComments] = useState([]);
+  const [items, setItems] = useState([]);
+
+  const loadItems = () => {
+    axios.get(`${myDogServerBaseURL}/breed/hound/list`).then((response) => {
+      console.log("response");
+      console.log(response);
+      setItems(response.data.message);
+    });
+  };
+
+  useEffect(() => {
+    dialog = document.querySelector(".dialog");
+    dialogMask = document.querySelector(".dialog__mask");
+    dialogWindow = document.querySelector(".dialog__window");
+  });
```

Comment:
useEffect will get called every time render is which I don't think is necessary.

If you can get away with just calling it once (the way componentDidMount works for classes) you can pass it a second argument of an empty array.

If you want to call this every time a certain state updates you can add those states into the array.

Reply:
Done